tools/clang/tools/dxcompiler/dxcutil.cpp

```diff
@@ -78,16 +78,20 @@ class DxilCompilerLLVMModuleOutput {
     CComPtr<AbstractMemoryStream> pContainerStream;
     IFT(CreateMemoryStream(pMalloc, &pContainerStream));
     SerializeDxilContainerForModule(&m_llvmModule->GetOrCreateDxilModule(),
-                                    pModuleBitcode, pContainerStream, Flags);
+                                    pModuleBitcode, pContainerStream, m_debugName, Flags);
 
     pDxilContainerBlob.Release();
     IFT(pContainerStream.QueryInterface(&pDxilContainerBlob));
   }
+  void SetDebugName(llvm::StringRef DebugName) {
+    m_debugName = DebugName;
```

Comment:
It currently won't cause issues, but it'd be safer to own the string here, so the caller doesn't have to worry about keeping it alive.